crates/holochain/src/core/ribosome/host_fn/send_remote_signal.rs

```diff
@@ -62,25 +65,49 @@ pub fn send_remote_signal(
                             nonce,
                             expires_at,
                         };
-                        let potentially_signature = zome_call_unsigned.provenance.sign_raw(call_context.host_context.keystore(), match zome_call_unsigned.data_to_sign() {
-                            Ok(to_sign) => to_sign,
+                        let zome_call_param_bytes = match zome_call_params.serialize() {
+                            Ok(bytes) => bytes,
                             Err(e) => {
-                                tracing::info!("Failed to serialize zome call for signal because of {:?}", e);
+                                tracing::info!(
+                                    "Failed to serialize zome call for signal because of {:?}",
+                                    e
+                                );
                                 return;
                             }
-                        }).await;
-
-                        match potentially_signature {
-                            Ok(signature) => to_agent_list.push((signature, agent)),
+                        };
+                        // Signature is generated for the hash of the serialized bytes.
+                        let bytes_hash = blake2b_256(&zome_call_param_bytes);
```

Comment:
Why is this done manually, isn't there a DRY helper function?

Reply:
There isn't. I combined it with the serialization function, as they were always used together.